contrib/format-hdf5/src/main/java/org/apache/drill/exec/store/hdf5/HDF5BatchReader.java

```diff
@@ -323,53 +328,20 @@ private void openFile(FileSchemaNegotiator negotiator) throws IOException {
     InputStream in = null;
     try {
       in = negotiator.fileSystem().openPossiblyCompressedStream(split.getPath());
-      IHDF5Factory factory = HDF5FactoryProvider.get();
-      inFile = convertInputStreamToFile(in);
-      hdf5Reader = factory.openForReading(inFile);
+      hdfFile = HdfFile.fromInputStream(in);
```

Comment:
Interesting.  The BatchReader has a `close()` method which is executed when Drill is finished reading the file.  Is there a method in JHDF to purge the temp files, or would you suggest using a different method to load the file?

https://github.com/apache/drill/blob/5729e1f921eaceac8647ee6eee24a4c2bb31e21e/contrib/format-hdf5/src/main/java/org/apache/drill/exec/store/hdf5/HDF5BatchReader.java#L1229-L1236

Reply:
For now you could use `hdfFile.getFile().delete()` after the `hdfFile.close()` method has been called probably via `AutoCloseable`. I will do this clean up automatically in https://github.com/jamesmudd/jhdf/issues/262

Reply:
> For now you could use `hdfFile.getFile().delete()` after the `hdfFile.close()` method has been called probably via `AutoCloseable`. I will do this clean up automatically in [jamesmudd/jhdf#262](https://github.com/jamesmudd/jhdf/issues/262)

@jamesmudd, 
Would you recommend using the loaders to load the file into memory instead of the current implementation?

Reply:
I guess it depends what file sizes you are expecting. If your files will fit in memory then it's probably a better option.

Reply:
I put comments in the code to note this as a possible improvement.  My thought is to check the file size and if it exceeds a certain size, use the current implementation and if not, read it from memory.

Reply:
I also added code to delete the temp file on close.

Reply:
Choosing the approach depending on the file size seems like a good idea. I will open a PR here when this auto-deletion support is in jhdf but for now this will be fine.